pkg/controller/deployment/deployment_controller.go

```diff
@@ -424,6 +424,27 @@ func (dc *DeploymentController) syncDeployment(key string) error {
 	return fmt.Errorf("unexpected deployment strategy type: %s", d.Spec.Strategy.Type)
 }
 
+// syncPausedDeployment allows scaling of the deployment (either done manually
+// or by autoscalers) to propagate to the active rc of the deployment.
+func (dc *DeploymentController) syncPausedDeployment(deployment extensions.Deployment) error {
+	newRC, err := deploymentutil.GetNewRCFromList(deployment, dc.client,
+		func(namespace string, options api.ListOptions) ([]api.ReplicationController, error) {
+			return dc.rcStore.List()
+		})
+	if err != nil {
+		return err
+	}
+	if newRC == nil {
+		// There is no replication controller that matches the intent of
+		// this deployment which means that either the deployment has never
+		// run (created as paused), or it was switched to paused and has been
+		// modified since. Such is life.
+		return nil
+	}
+	_, err = dc.scaleRCAndRecordEvent(newRC, deployment.Spec.Replicas, deployment)
```

Comment:
Which makes me think that we should implement canary checks. Also regarding test deployments see https://github.com/openshift/origin/pull/6930.